lumigator/jobs/inference/inference.py

```diff
@@ -121,16 +123,21 @@ def run_inference(config: InferenceJobConfig, api_key: str | None = None) -> Pat
 
     artifacts = InferenceJobOutput.model_validate(output)
 
+    all_metrics_present = all(p.metrics for p in prediction_results)
+
+    if any(p.metrics for p in prediction_results) != all_metrics_present:
+        raise ValueError("Prediction result 'metrics' must be present in ALL results or NONE, but not in SOME.")
+
+    results = JobOutput(artifacts=artifacts, parameters=config)
+
```

Comment:
This comment is almost certainly scope creep, but I wonder if the creation of this avg_results and artifact json file should get extracted out to a new class or function..

Reply:
I've pulled out the average calculations